base/complex.jl

```diff
@@ -1019,3 +1019,11 @@ function complex(A::AbstractArray{T}) where T
     end
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
+
+function complex(A::AbstractArray{T1}, B::AbstractArray{T2}) where {T1, T2}
```

Comment:
Just `A::AbstractArray{<:Real}, B::AbstractArray{<:Real}` should be good here, unless we want to support missing data.   Since `complex` does not support `missing` for the scalar case, however, I think adding missing support should happen in a separate PR that generalizes both vector and scalar cases.

Reply:
See above for a greatly simplified definition.